CS: update ruleset with recommended custom properties for improved checking

jrfnl opened the pull request:
Also: use whitelist comment instead of blanket code standards ignore tags.

GaryJones commented on phpcs.xml:
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0"?>
```
This file should be called `phpcs.xml.dist`, and `phpcs.xml` should be added to a `.gitignore`.

jrfnl replied:
Could you elaborate why ?

This is the project standard to use and individual contributors should not overrule it with their own `phpcs.xml` file.

GaryJones replied:
Because we're distributing it, but others may want to override this locally with their own PHPCS - maybe to test it against the next version of WPCS or some other reason.

You only need to check https://github.com/squizlabs/PHP_CodeSniffer itself to see `.dist` versions of config files, even if the project has it's own standard to use.

jrfnl replied:
I've made the change as I don't necessarily disagree with your suggestion.

> Because we're distributing it, but others may want to override this locally with their own PHPCS

As it's a library, people shouldn't adjust the code - even for style - when using it.

If it's being used in a larger project, PHPCS will use the `phpcs.xml(.distr)` file in the root - or directories above this - of the larger project anyway.
It will not respect `phpcs.xml` files in directories below the root of the project, so having a `phpcs.xml` in a subdirectory (`/tgmpa`) will not affect the CS run.

> - maybe to test it against the next version of WPCS or some other reason.

The XML file is independent of version - apart from new properties which may be introduced -. What version is used to run it against depends on the version set up locally or in the test environment.

GaryJones commented on phpcs.xml:
```diff
@@ -14,4 +16,24 @@
+		</properties>
+	</rule>
+
+	<rule ref="WordPress.WP.DeprecatedFunctions">
```
This looks like an exact copy of the rule above.

jrfnl replied:
Sorry, goes to show that I'm working on too many PRs at the same time. The second one should be for `WordPress.WP.DeprecatedParameters`. Adjusting now.

jrfnl replied:
Grr.. and I forgot that that last sniff hasn't been merged yet... oh well.. adding it to a backwing branch with other changes for when WPCS 0.12.0 comes out ;-)

jrfnl committed:
CS: update ruleset with recommended custom properties for improved checking.

Also: use whitelist comment instead of blanket code standards ignore tags.

GaryJones commented on .gitignore:
```diff
@@ -0,0 +1,2 @@
+# folders and files to be ignored by git
+phpcs.xml
```
Trailing clear linespace required.

jrfnl replied:
Fixed

jrfnl committed:
CS: Rename `phpcs.xml` to `phpcs.xml.dist`

Allows developers to overrule the standard ruleset with a - stricter - personal ruleset.
